ablySpec/RestClient.swift

```diff
@@ -326,6 +327,82 @@ class RestClient: QuickSpec {
+                            let options = AblyTests.clientOptions()
+                            options.key = client.options.key
+
+                            // Expired token
+                            options.tokenDetails = ARTAuthTokenDetails(token: currentTokenDetails.token, expires: currentTokenDetails.expires?.dateByAddingTimeInterval(testTimeout), issued: currentTokenDetails.issued, capability: currentTokenDetails.capability, clientId: currentTokenDetails.clientId)
+
+                            options.authUrl = NSURL(string: "http://test-auth.ably.io")
+
+                            let rest = ARTRest(options: options)
+
+                            // Delay for token expiration
+                            delay(tokenParams.ttl) {
+                                // 40140 - token expired and will not recover because authUrl is invalid
+                                publishTestMessage(rest) { error in
+                                    expect(error).toNot(beNil())
+                                    done()
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            // RSC14d
+            it("should renew the token when it has expired") {
+                let client = ARTRest(options: AblyTests.commonAppSetup())
+                let auth = client.auth
+
+                let tokenParams = ARTAuthTokenParams()
+                tokenParams.ttl = 3.0 //Seconds
+
+                waitUntil(timeout: testTimeout) { done in
+                    auth.requestToken(tokenParams, withOptions: nil) { tokenDetails, error in
+                        if let e = error {
+                            XCTFail(e.description)
+                            done()
+                        }
+                        else if let currentTokenDetails = tokenDetails {
+                            let options = AblyTests.clientOptions()
+                            options.key = client.options.key
+
+                            // Expired token
+                            options.tokenDetails = ARTAuthTokenDetails(token: currentTokenDetails.token, expires: currentTokenDetails.expires?.dateByAddingTimeInterval(testTimeout), issued: currentTokenDetails.issued, capability: currentTokenDetails.capability, clientId: currentTokenDetails.clientId)
+
+                            let rest = ARTRest(options: options)
+                            rest.httpExecutor = mockExecutor
+
+                            // Delay for token expiration
+                            delay(tokenParams.ttl) {
+                                // 40140 - token expired and will resend the request
+                                publishTestMessage(rest) { error in
+                                    expect(mockExecutor.responses.first?.allHeaderFields["X-Ably-ErrorCode"] as? String).to(equal("40140"))
+                                    expect(error).to(beNil())
```

Comment:
Might be nice to check that the current token string is different to the previous one as well?